pkg/termcorder/termcorder.go

```
@@ -0,0 +1,210 @@
+		QuietMode:     q,
+		Append:        a,
+		Interactive:   interactive,
+		LogKeystrokes: k,
+		PrintHelp:     h,
+	}, options...)
+}
+
+//Start runs the tc.cmd in a pseudo-terminal and writes all output to tc.out
+func (tc *Termcording) Start() error {
+	if tc.Config.PrintHelp {
+		if tc.out == nil {
+			tc.out = os.Stdout
+		}
+		printHelp(tc)
+		return nil
+	}
+
+	if tc.cmd == nil {
+		tc.cmd = exec.Command(tc.Config.CmdName, tc.Config.CmdArgs...)
+	}
+
+	if !tc.Config.QuietMode {
+		fmt.Println("Starting recording session. CTRL-D to end.")
+		defer fmt.Printf("\nRecording session ended. Session saved to %s\n", tc.Config.Filename)
+	}
+
+	ptmx, restoreMode, err := ptmxFromCmd(tc.cmd, tc.Config.Interactive) //TODO convert to method.
+	if err != nil {
+		return err
+	}
+	defer func() {
+		restoreMode()
+		ptmx.Close()
+	}()
+
+	in := io.Writer(ptmx)
+	if tc.out == nil {
+		mode := os.O_TRUNC
+		if tc.Config.Append {
+			mode = os.O_APPEND
+		}
+		f, err := os.OpenFile(tc.Config.Filename, os.O_WRONLY|os.O_CREATE|mode, 0755)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		if tc.Config.LogKeystrokes {
+			in = io.MultiWriter(ptmx, f)
+		}
+		tc.out = io.MultiWriter(os.Stdout, f)
+	}
+
+	go func() {
+		io.Copy(in, os.Stdin)
+	}()
+
+	io.Copy(tc.out, ptmx)
+
+	return tc.cmd.Wait()
+}
+
+func ptmxFromCmd(c *exec.Cmd, interactive bool) (*os.File, func(), error) {
+	modeRestoreFn := func() {}
+
+	ptmx, err := pty.Start(c)
+	if err != nil {
+		return nil, nil, err
+	}
+	if interactive {
+		// Handle pty size.
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, syscall.SIGWINCH)
+		go func() {
+			for range ch {
+				if err := pty.InheritSize(os.Stdin, ptmx); err != nil {
+					log.Printf("error resizing pty: %s", err)
+				}
+			}
+		}()
+		ch <- syscall.SIGWINCH // Initial resize.
+
+		// Set stdin in raw mode.
+		oldState, err := term.MakeRaw(int(os.Stdin.Fd()))
+		if err != nil {
+			log.Fatalf("failed to set input to raw mode: %s", err)
```

Libraries shouldn't terminate the program; that right should be reserved to `main`. Instead, let's return the error.